[#6] Book 엔티티 구현

tinajeong opened the pull request:
<!--제목은 `[#이슈번호] 이슈 제목` 으로 작성한다.-->

### 🍀 목적
- Book 엔티티 구현

### 🌹 추가사항<!-- 있다면 적고 없다면 적지 않는다.-->
- Book.java
- BookTest.java
- BookObjectProvider.java (테스트 시 사용하기 위한 용도)

resolves #6 <!-- pr이 머지되면 이슈가 자동으로 close되게 합니다. 만약 자동 close를 하지 않고 이슈만 링크한다면 resolves를 삭제한다.-->

tinajeong committed:
feat : Book 엔티티 정의 및 테스트

youngjijang commented on src/main/java/com/dadok/gaerval/domain/book/entity/Book.java:
```diff
@@ -0,0 +1,135 @@
+	private String apiProvider;
+
+	@Builder(builderMethodName = "requiredBuilder")
+	public Book(Long id, String title, String author, String isbn, String contents, String url,
```
id가 제외되어는게 좋을 것 같아요

youngjijang commented on src/test/java/com/dadok/gaerval/testutil/BookObjectProvider.java:
```diff
@@ -0,0 +1,19 @@
+
+	public static Book createRequiredFieldBook() {
+		return  Book.requiredBuilder()
+			.id(1L)
```
Id는  ReflectionsTestUtils로 지정해주기

youngjijang commented on src/main/java/com/dadok/gaerval/domain/book/entity/Book.java:
```diff
@@ -0,0 +1,135 @@
+		validateImageKey(imageKey);
+		validateApiProvider(apiProvider);
+	}
+
```
정적 팩토리 메소드 추가해주세요~

tinajeong committed:
refact : Book 엔티티
- builder 삭제
- 생성자 호출하는 방식으로 코드 중복 줄이기
- optional 필드 테스트에 추가
- 도서 엔티티 상태 변경 테스트 추가

devYSK commented on src/main/java/com/dadok/gaerval/domain/book/entity/Book.java:
```diff
@@ -0,0 +1,137 @@
+		validateApiProvider(apiProvider);
+	}
+
+	protected Book(String title, String author, String isbn, String contents, String url,
```
good
